src/app/modules/react-migration/react-migration.controller.ts

```diff
@@ -0,0 +1,118 @@
+import path from 'path'
+
+import { FormAuthType } from '../../../../shared/types'
+import config from '../../config/config'
+import { ControllerHandler } from '../core/core.types'
+import * as FormService from '../form/form.service'
+import * as PublicFormController from '../form/public-form/public-form.controller'
+import { RedirectParams } from '../form/public-form/public-form.types'
+import * as HomeController from '../home/home.controller'
+
+const RESPONDENT_COOKIE_NAME = 'v2-respondent-ui'
+const ADMIN_COOKIE_NAME = 'v2-admin-ui'
+
+export type SetEnvironmentParams = {
+  ui: 'react' | 'angular'
+}
+
+export const RESPONDENT_COOKIE_OPTIONS = {
+  httpOnly: true,
+  sameSite: 'strict' as const,
+  secure: !config.isDev,
+}
+
+export const ADMIN_COOKIE_OPTIONS = {
+  httpOnly: true,
+  maxAge: 365 * 24 * 60 * 60, // 1 year
+  sameSite: 'strict' as const,
+  secure: !config.isDev,
+}
+
+const serveFormReact: ControllerHandler = (_req, res) => {
+  const reactFrontendPath = path.resolve('dist/frontend')
+  res.sendFile(path.join(reactFrontendPath, 'index.html'))
+}
+
+const serveFormAngular: ControllerHandler<
+  RedirectParams,
+  unknown,
+  unknown,
+  Record<string, string>
+> = (req, res, next) => {
+  PublicFormController.handleRedirect(req, res, next)
+}
+
+export const serveForm: ControllerHandler<
+  RedirectParams,
+  unknown,
+  unknown,
+  Record<string, string>
+> = async (req, res, next) => {
+  const formResult = await FormService.retrieveFormKeysById(req.params.formId, [
+    'authType',
+  ])
+  let showReact: boolean | undefined = undefined
+  let isSPCP = true
+
+  if (!formResult.isErr()) {
+    isSPCP = [FormAuthType.SP, FormAuthType.CP, FormAuthType.MyInfo].includes(
```

Comment:
change this conditional to isSPCP = formResult.value.authType === FormAuthType.Nil ?

Reply:
https://github.com/opengovsg/FormSG/pull/3750/commits/b5ceec2aa436830782f93e84ebed30d295fc434a